great_expectations/rule_based_profiler/domain_builder/domain_builder.py

```diff
@@ -55,6 +55,19 @@ def get_domains(
         """
         return self._get_domains(variables=variables)
 
+    @property
+    @abstractmethod
+    def domain_type(self) -> Union[str, MetricDomainTypes]:
+        pass
+
+    @property
+    def batch_request(self) -> Optional[Union[BatchRequest, RuntimeBatchRequest, dict]]:
+        return self._batch_request
+
+    @property
+    def data_context(self) -> "DataContext":  # noqa: F821
+        return self._data_context
```

Comment:
Do we need to expose this behavior in a property? This will make refactoring the context out a bit harder.

Reply:
Agreed. Would like to know if this is totally necessary for domain builder

Reply:
@cdkini @Shinnnyshinshin I am not sure if we must expose it (I know that subclasses use it, but that may not be enough of a justification).  Since I only moved things here (for a better organization/readability), I would propose that doing anything about this is deferred to a concerted effort.  Thanks.

Reply:
Since you just moved things, I'm okay with this.